Automattic/wp-calypso
# Gutenberg: Reorganize and cleanup Markdown block

This PR is a set of janitorial tasks that I've been wanting to push on the Markdown block:

* Rename all the files to be more consistent with naming conventions in the Gutenberg core block library:
  * `jetpack-markdown-block-editor.jsx` → `edit.jsx`
  * `jetpack-markdown-block-save.jsx` → `save.jsx`
  * `jetpack-markdown-block.jsx` → `editor.jsx`
  * `/components/markdown-renderer.jsx` → `renderer.jsx`
  * `/markdown-editor.scss` → `style.scss`
* Rename the `JetpackMarkdownBlockEditor` component to `MarkdownEdit`.
* Get rid of an unnecessary function call in the `render` method of `edit` in favor of using declarative syntax.
* Simplify the `renderer` by converting the functional component to an arrow function.
* Simplify `save` by converting the functional component to an arrow function.

No visual, functional or behavioral changes are introduced by this PR.

To test:
* Checkout this branch on your local Calypso
* Checkout the latest Jetpack on your local machine.
* Setup Jetpack's Docker environment.
* Add the following as a mu-plugin in the Jetpack docker env:
  * `add_filter( 'jetpack_gutenberg', '__return_true', 10 );`
  * `add_filter( 'jetpack_gutenberg_cdn', '__return_false', 10 );`
* Make sure Jetpack is connected on your install.
* Make sure you got the latest Gutenberg installed.
* Run the following command in Calypso (where `PATH_TO_JETPACK` is the path to your Jetpack repo) to build the Jetpack preset of blocks:

```
npm run sdk -- gutenberg \
--editor-script=client/gutenberg/extensions/presets/jetpack/editor.js \
--output-dir=/PATH_TO_JETPACK/_inc/blocks \
--output-editor-file=jetpack-editor \
```

* Write a post, insert a Markdown block.
* Play the block with both no content and with content in both the Markdown and the Preview tabs - there should be no visual/functional changes.
